Application/EdFi.Security.DataAccess/Repositories/ISecurityRepository.cs

```diff
@@ -4,35 +4,39 @@
 // See the LICENSE and NOTICES files in the project root for more information.
 
 using System.Collections.Generic;
+using Autofac.Extras.DynamicProxy;
+using Castle.DynamicProxy;
 using EdFi.Security.DataAccess.Models;
 
-namespace EdFi.Security.DataAccess.Repositories
-{
-    public interface ISecurityRepository
-    {
-        Action GetActionByHttpVerb(string httpVerb);
+namespace EdFi.Security.DataAccess.Repositories;
 
-        Action GetActionByName(string actionName);
+/// <summary>
+/// Defines higher-level methods for obtaining security metadata for making authorization decisions.
+/// </summary>
+/// <remarks>Implementations of this interface must be configured with a named <see cref="IInterceptor" /> registration of "cache-security".</remarks>
+[Intercept("cache-security")]
+public interface ISecurityRepository
+{
+    Action GetActionByName(string actionName);
 
-        AuthorizationStrategy GetAuthorizationStrategyByName(string authorizationStrategyName);
+    AuthorizationStrategy GetAuthorizationStrategyByName(string authorizationStrategyName);
 
-        IEnumerable<ClaimSetResourceClaimAction> GetClaimsForClaimSet(string claimSetName);
+    IList<ClaimSetResourceClaimAction> GetClaimsForClaimSet(string claimSetName);
```

Comment:
Changing `IEnumerable` to `IList`: This is probably safe to do, but I doubt that this change is necessary for the 6.x codebase. Best to keep the original interface though, and change back to `IEnumerable`, unless you have a .NET 8 warning or code suggestion. If you do have such a warning, then `IReadOnlyList` is probably a better choice: based on the names, this code is only for retrieving data, not for changing it, so a read only list would be appropriate. I also looked at `ClaimsIdentityProvider` in `EdFi.Ods.Api` and confirmed that `IReadOnlyList` will work there.